balboa-service-jms/src/main/java/com/socrata/balboa/jms/activemq/Consumer.java

```diff
@@ -37,28 +39,34 @@ public void onMessage(Message payload) {
         long start = System.currentTimeMillis();
         String messageText = null;
         try {
-            TextMessage text = (TextMessage) payload;
-            messageText = text.getText();
+            // Before we consider chunking a large message, check if we even need to write to send out the message
+            if (stopWrites) {
+                session.commit();
```

Comment:
is this even necessary? if no changes are being made to Cassandra then a commit isn't necessary? or are commits something you should regularly do, even during a NOOP?

Reply:
The session is related to ActiveMQ, not Cassandra. I believe we should commit here, else (afaik) the message won't be acknowledged. https://activemq.apache.org/how-do-transactions-work

Reply:
Ah ok good, I was wondering how messages were acked/nacked here